### PR TITLE
Parameter Model Improvements

### PR DESCRIPTION
**Pull request by jmchilton**
See individual commits - all enhancements stemming from downstream work on validating workflows and tool test cases with the models introduced in #18524 as part of the broader structured tool state work #17393 / https://docs.google.com/document/d/1HQOLpLN54CjrB-wbD463XqzvUm-dNB8vTXUFBExh_2o/edit. 

## How to test the changes?
(Select all options that apply)
- [x] I've included appropriate [automated tests](https://docs.galaxyproject.org/en/latest/dev/writing_tests.html).

## License
- [x] I agree to license these and all my past contributions to the core galaxy codebase under the [MIT license](https://opensource.org/licenses/MIT).

### CODE REVIEW

File: test/functional/tools/parameters/gx_select.xml [hunk 1]
```diff
@@ -23,5 +23,15 @@ echo '$parameter' >> '$output'
                 </assert_contents>
             </output>
         </test>
+        <test>
+            <!-- Selects have an implicit first option default in tool tests.
+                 Do they in the actual API also?
+            -->
```
Comment (mvdbeek) [inline test/functional/tools/parameters/gx_select.xml:29]:
```suggestion
            <!-- Selects implicitly default to the first option as the default option.
                    Additional verification in test_select_first_by_default API test.
            -->
```
I wonder if this is good behavior overall ? Say we're missing any valid options, when would we catch this during validation ?

File: lib/galaxy/tool_util/parameters/models.py [hunk 2]
```diff
@@ -478,9 +479,112 @@ def pydantic_template(self, state_representation: StateRepresentationT) -> Dynam
...
+    @property
+    def py_type(self) -> Type:
+        return StrictInt
+
+    def pydantic_template(self, state_representation: StateRepresentationT) -> DynamicModelInformation:
+        return dynamic_model_information_from_py_type(self, self.py_type)
+
+    @property
+    def request_requires_value(self) -> bool:
+        return False
```
Comment (mvdbeek) [inline lib/galaxy/tool_util/parameters/models.py:587]: I'm at least a little surprised 😅

**Review by mvdbeek — APPROVED** [review]
Looks great so far!

File: lib/galaxy/tool_util/parameters/models.py (hunk 2 above)
Comment (jmchilton) [reply lib/galaxy/tool_util/parameters/models.py:587]: Just a first cut - I think it has slightly different behavior based on how it is configured. The tests will keep coming.

File: test/functional/tools/parameters/gx_select.xml (hunk 1 above)
Comment (jmchilton) [reply test/functional/tools/parameters/gx_select.xml:29]: It is all green enough - I'll fix this comment as the first thing I do in my next model rev - I am sure a new PR will be open this week. Thanks

**Pull request merged**